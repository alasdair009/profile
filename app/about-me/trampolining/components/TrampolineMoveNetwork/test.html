<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
