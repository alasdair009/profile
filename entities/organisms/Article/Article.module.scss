@use "../../../app/styles/colors";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/helpers/rem";

.root {
  margin: 0 auto;
  max-width: rem.toRem(dimensions.$globalContentMaxWidth);
  position: relative;
  width: 100%;
}

.articleHeading {
  max-width: dimensions.getSize(s512);
}

.articleDate {
  display: block;
  font-style: italic;
  margin: 0 auto;
  max-width: rem.toRem(dimensions.$globalTextMaxWidth);
  text-align: center;
}

.articleFigure {
  box-sizing: border-box;
  height: rem.toRem(320);
  margin: 0 auto;
  max-width: calc(100% - #{dimensions.getSize(s16)});
  position: relative;

  @media (map-get(dimensions.$device, large)) {
    max-width: rem.toRem(dimensions.$globalTextMaxWidth);
  }
}

.articleImage {
  box-shadow:
    -(dimensions.getSize(s48)) -(dimensions.getSize(s48))
      0 -(dimensions.getSize(s40)) colors.$colorGreenGrass,
    dimensions.getSize(s48) dimensions.getSize(s48) 0 -(dimensions.getSize(s40))
      colors.$colorGreenGrass;
  margin: dimensions.getSize(s8);
  max-width: calc(100% - #{dimensions.getSize(s16)});
}

.content {
  margin: 0 auto;
  max-width: rem.toRem(dimensions.$globalTextMaxWidth);
  overflow: hidden;
  padding: dimensions.getSize(s8);
  position: relative;
  width: 100%;
  word-break: break-word;

  iframe {
    max-width: rem.toRem(800);
  }
}
