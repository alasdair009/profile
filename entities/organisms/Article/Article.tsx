import {
  ArticleDate,
  ArticleFigure,
  ArticleHeading,
  ArticleImage,
  Content,
  Root,
} from "./styles";
import { HTMLAttributes, ReactNode } from "react";
import { Spacer } from "@/entities";

type ArticleProps = {
  heading: string;
  image: string;
  date: Date;
  children: ReactNode;
} & HTMLAttributes<HTMLDivElement>;

const nthNumber = (number: number) => {
  if (number > 3 && number < 21) return "th";
  switch (number % 10) {
    case 1:
      return "st";
    case 2:
      return "nd";
    case 3:
      return "rd";
    default:
      return "th";
  }
};

export const getNiceDate = (date: Date) => {
  const dayNumber = date.toLocaleDateString("en-GB", {
    day: "numeric",
  });

  return `${dayNumber}${nthNumber(parseInt(dayNumber))} ${date.toLocaleDateString("en-GB", { year: "numeric", month: "long" })}`;
};

/**
 * Large content box with heading
 */
export function Article({
  heading,
  image,
  date,
  children,
  ...rest
}: ArticleProps) {
  const dayNumber = date.toLocaleDateString("en-GB", {
    day: "numeric",
  });
  return (
    <Root data-testid={Article.name} {...rest}>
      <ArticleHeading align="center">{heading}</ArticleHeading>
      <ArticleDate dateTime={date.toISOString()}>
        {getNiceDate(date)}
      </ArticleDate>
      <Spacer multiplier={6} />
      <ArticleFigure>
        <ArticleImage
          src={image}
          alt={heading}
          fill={true}
          style={{ objectFit: "cover" }}
          priority={true}
        />
      </ArticleFigure>
      <Spacer multiplier={8} />
      <Content>{children}</Content>
    </Root>
  );
}
