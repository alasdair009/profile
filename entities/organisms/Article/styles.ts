"use client";

import styled from "styled-components";
import { colors, device, globalContentMaxWidth, globalTextMaxWidth, Heading, sizes } from "@/entities";
import { rem } from "polished";
import Image from "next/image";

export const Root = styled.article`
  margin: 0 auto;
  max-width: ${rem(globalContentMaxWidth)};
  position: relative;
  width: 100%;
`;

export const ArticleHeading = styled(Heading)`
  max-width: ${sizes.s512.rem};
`;

export const ArticleDate = styled.time`
  display: block;
  font-style: italic;
  margin: 0 auto;
  max-width: ${rem(globalTextMaxWidth)};
  text-align: center;
`;

export const ArticleFigure = styled.figure`
  box-sizing: border-box;
  height: ${rem(320)};
  margin: 0 auto;
  max-width: calc(100% - ${sizes.s16.rem});
  position: relative;

  @media (${device.large}) {
    max-width: ${rem(globalTextMaxWidth)};
  }
`;

export const ArticleImage = styled(Image)`
  box-shadow:
    -${sizes.s48.rem} -${sizes.s48.rem} 0 -${sizes.s40.rem} ${colors.greenGrass},
    ${sizes.s48.rem} ${sizes.s48.rem} 0 -${sizes.s40.rem} ${colors.greenGrass};
  margin: ${sizes.s8.rem};
  max-width: calc(100% - ${sizes.s16.rem});
`;

export const Content = styled.div`
  margin: 0 auto;
  max-width: ${rem(globalTextMaxWidth)};
  overflow: hidden;
  padding: ${sizes.s8.rem};
  position: relative;
  width: 100%;
  word-break: break-word;

  iframe {
    max-width: ${rem(800)};
  }
`;
