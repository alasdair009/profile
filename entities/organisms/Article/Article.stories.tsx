import { Meta, StoryObj } from "@storybook/react";
import { Article, getNiceDate } from "./Article";
import amLogo from "../../assets/am.svg";
import { expect, within } from "@storybook/test";

const meta: Meta<typeof Article> = {
  component: Article,
  args: {
    heading: "Article heading",
    image: amLogo,
    date: new Date("2024-01-01"),
    children: "hello world",
  },
};
export default meta;

export const Default: StoryObj<typeof Article> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const articleElement: HTMLDivElement = canvas.getByTestId(Article.name);
    const dateElement: HTMLTimeElement = canvas.getByText(
      getNiceDate(args.date)
    );
    const imageElement: HTMLImageElement = canvas.getByAltText(args.heading);

    await expect(articleElement).toBeInTheDocument();
    await expect(canvas.getByText(args.heading)).toBeInTheDocument();
    await expect(imageElement).toBeInTheDocument();
    await expect(imageElement.src).toContain(amLogo.src);
    await expect(dateElement).toBeInTheDocument();
    await expect(dateElement).toHaveAttribute(
      "datetime",
      args.date.toISOString()
    );
    await expect(canvas.getByText(args.children as string)).toBeInTheDocument();
  },
};
