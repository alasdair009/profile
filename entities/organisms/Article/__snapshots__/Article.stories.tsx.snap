// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/Article Default play-test 1`] = `
<article data-testid="Article"
         class="sc-eowDPD iMDoqX"
>
  <h1 data-testid="Heading"
      class="sc-hfvVTD dcGPuN sc-iAlELC eeFioF"
  >
    Article heading
  </h1>
  <time datetime="2024-01-01T00:00:00.000Z"
        class="sc-oeqTF lmvAm"
  >
    1st January 2024
  </time>
  <br data-testid="Spacer"
      class="sc-pFPEP jOJHEi"
  >
  <figure class="sc-ezTrPE jVCBys">
    <img alt="Article heading"
         fetchpriority="high"
         decoding="async"
         data-nimg="fill"
         class="sc-drnuxz bXzTOE"
         sizes="100vw"
         srcset="/static/media/entities/assets/am.svg?w=640&amp;q=75 640w, /static/media/entities/assets/am.svg?w=750&amp;q=75 750w, /static/media/entities/assets/am.svg?w=828&amp;q=75 828w, /static/media/entities/assets/am.svg?w=1080&amp;q=75 1080w, /static/media/entities/assets/am.svg?w=1200&amp;q=75 1200w, /static/media/entities/assets/am.svg?w=1920&amp;q=75 1920w, /static/media/entities/assets/am.svg?w=2048&amp;q=75 2048w, /static/media/entities/assets/am.svg?w=3840&amp;q=75 3840w"
         src="/static/media/entities/assets/am.svg?w=3840&amp;q=75"
         style="position: absolute; height: 100%; width: 100%; inset: 0px; object-fit: cover; color: transparent;"
    >
  </figure>
  <br data-testid="Spacer"
      class="sc-pFPEP jOJJCx"
  >
  <div class="sc-hDcvty hROsLr">
    hello world
  </div>
</article>
`;
